Dev/nick9500/graphs

Nick9500 opened the pull request:
References EPIC #20

GurkomalSRao committed:
Chart Testing with minor changes to myData Class

Junit test for constructing a graph/chart

implementation of setting and getting url strings for MyData class

Nick9500 committed:
Added JSON Parsing Algorithm to retrieve 52wk closing values
References issue #20

Nick9500 committed:
Ninja Fix

Nick9500 committed:
JsonObjectRoot needs to not be null

Todo commit

Nick9500 committed:
Added StockChart Views On-Click Events

In this commit, stock chart popups were introduced using AlphaVantage data for the past 52 weeks' closing values. Charting imagery was using image-charts.com. Users may now tap on stocks in the widget and receive a popup showing the stocks 52wk performance instantly. This references issue #20.
Commit was done by Gurk, Jeff, Nick, and Github Raj.

GurkomalSRao committed:
editing Test file

everlybolivar committed:
Added Buttons to Graphs - #37
Task 1 Completed
Buttons still have no functionality
UI Fix: Graphs no longer span the whole screen

Nick9500 committed:
UI - Adding Functionality to Graph Buttons

In reference to issues #37, #20, Bev and I have added actual functionality to the ChartActivity view that pops up once you select a stock from the list. Users can now click on one of 3 buttons: 7 Day, 52wk, 12 Mths to change the time interval of the stock graph being displayed in the popup. Minor bugs to fix include a bouncing menu bar and the fact that you have to close the popup in order to see the changes take place.

everlybolivar committed:
Refinement #37, #20
Fixed enhancement so that once Time interval button is clicked graph refreshes at once. User no longer has to exit out of graph and reclick the stock to get updated graph with selected time interval.
Also re-stylized some code to make it more clear within MyData.java

Nick9500 committed:
Rebase on Master

Commit purely to make this branch up-to-date with master along with integrated changes.
Reference to #20, #37

Nick9500 committed:
Ninja fix for tests

everlybolivar committed:
Reformatted code style to match the rest of master - #45
In order to remain consistent, the code was reformated per our chosen style to match that of master once branch is ready to be merged

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -24,18 +38,92 @@
         Bitmap bmImg = null;
+        int interval = GlobalWidgetData.getInterval();
         int position = getIntent().getIntExtra(WidgetProviderBase.ROW_POSITION, 0);
+        //MyData.setInterval(2); /*= getIntent().getIntExtra(WidgetProviderBase.INTERVAL, 2);*/
```
remove unnecessary comment

Nick9500 replied:
fixed

jeffersoncasimir commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -24,18 +38,92 @@
+        super.onResume();
+        int interval = GlobalWidgetData.getInterval();
+        int position = getIntent().getIntExtra(WidgetProviderBase.ROW_POSITION, 0);
+        Button but7day = (Button) findViewById(R.id.but_7day);
```
I think "btn" is a better and more universal acronym than "but" for buttons

jeffersoncasimir replied:
Note: Every request refers to the last line shown (73 in this case)

Nick9500 replied:
fixed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -24,18 +38,92 @@
     @Override
     public void onCreate(Bundle savedInstanceState) {
         URL url = null;
         Bitmap bmImg = null;
```
url and bmImg are not being used so should be removed

Nick9500 replied:
fixed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -1,21 +1,35 @@
 package nitezh.ministock.activities;
 
 import android.app.Activity;
+import android.app.PendingIntent;
```
PendingIntent up to Intent imports are not used and should be removed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -1,21 +1,35 @@
 import android.text.Spanned;
 import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
```
`import android.view.ViewGroup;` not being used

Nick9500 replied:
fixed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -1,21 +1,35 @@
 import android.widget.ImageView;
 import android.widget.TextView;
 
+import com.google.gson.JsonObject;
```
`import com.google.gson.JsonObject;` not used

Nick9500 replied:
fixed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -1,21 +1,35 @@
+import java.util.List;
 
 import nitezh.ministock.R;
+import nitezh.ministock.WidgetProvider;
```
`import nitezh.ministock.WidgetProvider;` not used

Nick9500 replied:
fixed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -1,21 +1,35 @@
 import nitezh.ministock.R;
+import nitezh.ministock.WidgetProvider;
 import nitezh.ministock.activities.widget.WidgetProviderBase;
+import nitezh.ministock.activities.widget.WidgetRow;
```
` import nitezh.ministock.activities.widget.WidgetProviderBase; import nitezh.ministock.activities.widget.WidgetRow;`

are not used

Nick9500 replied:
fixed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -1,21 +1,35 @@
 
 import android.app.Activity;
+import android.app.PendingIntent;
+import android.appwidget.AppWidgetManager;
```
`import android.app.PendingIntent;
import android.appwidget.AppWidgetManager;
import android.content.Context;
import android.content.Intent;`

not used and should be removed

Nick9500 replied:
fixed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -24,18 +38,92 @@
+        String intervalStr = intervalSwitcher(interval);
+        Log.i("urltext", "https://www.alphavantage.co/query?function=TIME_SERIES_" + intervalStr + "_ADJUSTED&symbol="
+                + symbol + "&apikey=" + alphavantagekey);
+
```
remove unnecessary white space

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/ChartActivity.java:
```diff
@@ -24,18 +38,92 @@
+                startActivity(getIntent());
+                //recreate();
+
+                // Log.i("TEST PRINT", "WE CLICKED 7 DAY");
```
remove `// Log.i("TEST PRINT", "WE CLICKED 7 DAY");`
and all other logs below that are commented out

Nick9500 replied:
fixed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/GlobalWidgetData.java:
```diff
@@ -1,20 +1,37 @@
 import nitezh.ministock.activities.widget.WidgetRow;
-
+import java.util.Map;
+import java.util.Set;
```
`import nitezh.ministock.activities.widget.WidgetRow;
import java.util.Map;
import java.util.Set;`

are not used and should be removed

Nick9500 replied:
fixed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/GlobalWidgetData.java:
```diff
@@ -23,4 +40,187 @@
+
+    // Methods for Graph Implementation
+    public static JsonObject getJsonObjectRoot(String sURL) {
+        JsonObject rootObj;
```
`public static JsonObject getJsonObjectRoot(String sURL)`
method never used and should be removed if we are not going to use it in the future

Nick9500 replied:
I'll keep this in for future use.

jeffersoncasimir commented on src/main/java/nitezh/ministock/activities/GlobalWidgetData.java:
```diff
@@ -23,4 +40,187 @@
+                    dayCounter++;
+                }
+            }
+            if (this.interval == 3) {
```
This should be an else-if or refactored to a switch for further flexibility, IMO

Nick9500 replied:
fixed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/GlobalWidgetData.java:
```diff
@@ -23,4 +40,187 @@
+        chxt = "chxt=x%2Cy";                                         //Specify X and Y Axes
+        chs = "chs=700x690";                                        //chart size
+        chtt = "chtt=Test%20Graph";                                 //Name of Graph
+        String chdVars = "";
```
`chtt = "chtt=Test%20Graph";                                 //Name of Graph`

this should be renamed to the name of the actual graph. the name "test graph" was chosen for test cases

Nick9500 replied:
Changed

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/MinistocksActivity.java:
```diff
@@ -38,6 +38,7 @@ of this software and associated documentation files (the "Software"), to deal
         super.onCreate(savedInstanceState);
+        GlobalWidgetData.setInterval(2);
         setContentView(R.layout.ministocks);
         Spanned html = Html.fromHtml("Note: This is a widget only app.<br /><br />" + "<b>Getting started</b><br /><br />" + "First long-press an empty space on your Home screen to bring up the <i>Add to Home screen</i> menu.<br /><br />" + "Then select the Widgets option and finally, select the <i>MinistocksActivity</i> item from the list of widgets.<br /><br />" + "Multiple widgets can be added ...
```
`Spanned html = Html.fromHtml("Note: This is a widget only app.<br /><br />" + "<b>Getting started</b><br /><br />" + "First long-press an empty space on your Home screen to bring up the <i>Add to Home screen</i> menu.<br /><br />" + "Then select the Widgets option and finally, select the <i>MinistocksActivity</i> item from the list of widgets.<br /><br />" + "Multiple widgets can be added, as each widget stores its own data.<br /><br />" + "<b>Press Home or Back to close.</b><br />");`

is not styled properly. very difficult to read and should be properly formatted to fit the screen

Nick9500 replied:
fixed\

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/widget/Bonobo_widget_data_provider.java:
```diff
@@ -1,5 +1,9 @@
+import android.annotation.SuppressLint;
 import android.appwidget.AppWidgetManager;
 import android.content.Context;
 import android.content.Intent;
```
`import android.annotation.SuppressLint; import java.util.ArrayList;
import java.util.List;`

not being used

Nick9500 replied:
fixed

jeffersoncasimir commented on src/main/java/nitezh/ministock/activities/GlobalWidgetData.java:
```diff
@@ -23,4 +40,187 @@
+                int dayCounter = 0;
+
+                for (Map.Entry<String, JsonElement> entry : entries) {
+                    if (dayCounter <= 6) {
```
Suggestion: This logic should be extracted to a method since what follows is constantly re-used. Would improve readability and cohesion.

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/widget/Bonobo_widget_data_provider.java:
```diff
@@ -26,7 +30,6 @@
     Context mContext = null;
     private int mAppWidgetId;
 
```
`private int mAppWidgetId;`

can be converted to a local variable

Nick9500 replied:
Needs to stay where it is to ensure that it doesn't change over lifecycle.

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/widget/WidgetProviderBase.java:
```diff
@@ -51,8 +51,11 @@ of this software and associated documentation files (the "Software"), to deal
 
     public static String ROW_POSITION = "ROW_POSITION";
+    public static String INTERVAL = "2";
 
```
```
public static final String EXTRA_ITEM = "EXTRA_ITEM";
public static final String TOAST_ACTION = "TOAST_ACTION";
public static String INTERVAL = "2";
```

not being used and should be removed

Nick9500 replied:
fixed

GurkomalSRao commented on src/test/java/nitezh/ministock/ChartDataTesting/ChartDataTesting.java:
```diff
@@ -0,0 +1,82 @@
+        urlString = "https://image-charts.com/chart?"+cht+"&"+chd+chdVars+"&"+chds+"&chof=.png&"+chs+
+                "&chdls=000000&chco=F56991%2CFF9F80%2CFFC48C%2CD1F2A5%2CEFFAB4&"+chtt+"&"+chxt+"&chdlp=b&chf=bg%2Cs%2CFFFFFF&chbh=10&icwt=false";
+
+
```
unnecessary white space

GurkomalSRao commented on src/test/java/nitezh/ministock/ChartDataTesting/ChartDataTesting.java:
```diff
@@ -0,0 +1,82 @@
+            assertEquals(request.HTTP_OK, request.getResponseCode());
+
+
+            //chartActivity.onCreate(bundle);
```
`//chartActivity.onCreate(bundle);`
should be removed as well as the white space above it

Nick9500 replied:
fixed

GurkomalSRao reviewed (commented):
mostly formatting and removing unnecessary lines of code

everlybolivar reviewed (changes requested):
After reviewing and noting all the others results, I believe the reviewers touched on every single thing.  

I agree that simple clean up within the code and classes to ensure that the imports are absolutely necessary. Jeff brought up a very good suggestion in regards to either placing a switch case or creating a helper method for the logic of getting the button functionalities.

Nick9500 committed:
CodeReview Fixes

References issue #50 #20
Incorporated suggested changes from the codereview of the pullrequest for graph feature.

everlybolivar reviewed (approved):
Good, we'll just keep an eye out for the GetJsonObject method to make sure we use it.

gmacrodose committed:
Addressed a minor code review change request before accepting merge request #53
